csi/moac/doc/volume-operator.md

```diff
@@ -0,0 +1,62 @@
+# Volume Operator
+
+Volume operator is responsible for volume creation, destruction and recovery of
+volumes on storage nodes.  By volume, we mean "persistent volume" in K8s, which
+is dynamically created as a result of creating "persistent volume claim". A
+volume consists of a nexus and replica(s). Nexus is a logical volume, entry
+point for IOs, which are further dispatched to replicas - physical volumes,
+which allocate space from storage pools and can be accessed either locally (in
+the same process where nexus lives) or using remote storage protocol NVMEoF.
+
+The volume operator relies on following information when making decisions:
+
+* info from CSI requests (i.e. create volume parameters from CSI RPC method)
+* info obtained from storage nodes (i.e. list of created replicas and their parameters)
+* info obtained from msv k8s custom resources (i.e. was the volume mounted)
+
+When moac starts up it reads all available information from storage nodes and
+then tries to create/modify objects according to what it believes is the desired
+state. The source of the information about the desired are msv resources. For
```

Comment:
Need to add "state" -> "...about the desired **state** are msv resources."

Reply:
fixed. thanks